cross-lib/update_books.rb

```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env ruby
+
+require 'json'
+require 'set'
+
+# Files
+base_dir = 'cross-lib/book-data'
+out_file = "#{base_dir}/AUTO_books.txt"
+ubl = "#{base_dir}/USER_ubl.txt"
+`mkdir -p #{base_dir}; touch #{out_file}; touch #{ubl}`
+
+
+# Refresh ABL (bash)
+abl_hash = JSON.parse(
+  `curl -Ss https://raw.githubusercontent.com/openstax/content-manager-approved-books/main/approved-book-list.json`
+)
+
+books=Set[]
+# Write unapproved books
+if File.exist?(ubl)
+  File.readlines(ubl).each do |line|
+    books.add(line)
+  end
+end
+
+# Write ABL
+abl_hash['approved_books'].each do |book|
+  repo = book['repository_name']
+  book['versions'][0]['commit_metadata']['books'].each do |volume| 
```

Comment:
One problem I ran into here is that the first entry in the version array tends to be the oldest entry (although there is nothing that guarantees this). The book slug in older versions of books on the ABL does not necessarily exist in the newest version of the repository (example: american-government vs american-government-3e).

Maybe you could store the `commit_sha` and use that when building books instead of `main`. Additionally, if you only want to build the latest version of each book, you could sort the `versions` by their `commit_metadata.committed_at` field.

Reply:
thanks for the tip! i've gone with the latest version.

ultimately, i hope the slug will go away & books will be built by repo with this builder (pending [this](https://github.com/openstax/ce/issues/1971) 😅 ). but that can be a next pass